server/controllers/studentController.js

```diff
@@ -81,5 +94,48 @@ const updateStudent = async (req, res) => {
 
 }
 
+//generating token function
+const createToken = (_id) => {
```

Comment:
You coded this fn already within the company controller. I would recommend having a shared service/controller (so called authController) which handles all the signin & signup processes. If you need to differ between student & company login, provide an additional parameter to the corresponding login/register fn like "isCompany" of type boolean

Reply:
Yeah I do agree.. maybe i will do them later as ( optimization and enhancement of my code).